src/sparsezoo/v2/directory.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import pathlib
+import tarfile
+from typing import List, Optional
+
+from sparsezoo.utils.downloader import download_file
+from sparsezoo.v2.file import File
+
+
+__all__ = ["Directory"]
+
+
+class Directory(File):
+    """
+    Object that represents a directory.
+    Directory may also represent a tar file.
+
+    :param files: list of files contained within the Directory
+        (None for tar archive files).
+    :param name: name of the Directory
+    :param path: path of the Directory
+    :param url: url of the Directory
+    """
+
+    def __init__(
+        self,
+        name: str,
+        files: Optional[List[File]] = None,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+        # TODO: Adding unpack as a parameter, mainly because of testing
+        # limitations (e.g. I cannot unpack tar archives which I try to
+        # download from SparseZoo but they are not there yet).
+        # We may remove this argument later.
```

Comment:
can we detect this and set upack accordingly?

Reply:
Good idea, this should be beneficial.